cloud/pkg/devicecontroller/controller/upstream.go

```diff
@@ -111,7 +115,67 @@ func (uc *UpstreamController) updateDeviceStatus() {
 		case <-beehiveContext.Done():
 			klog.Info("Stop updateDeviceStatus")
 			return
-		case msg := <-uc.deviceStatusChan:
+		case msg := <-uc.deviceStatesChan:
+			klog.Infof("Message: %s, operation is: %s, and resource is: %s", msg.GetID(), msg.GetOperation(), msg.GetResource())
+			msgState, err := uc.unmarshalDeviceStatesMessage(msg)
+			if err != nil {
+				klog.Warningf("Unmarshall failed due to error %v", err)
+				continue
+			}
+			deviceID, err := messagelayer.GetDeviceID(msg.GetResource())
+			if err != nil {
+				klog.Warning("Failed to get device id")
+				continue
+			}
+			device, ok := uc.dc.deviceManager.Device.Load(deviceID)
+			if !ok {
+				klog.Warningf("Device %s does not exist in upstream controller", deviceID)
+				continue
+			}
+			cacheDevice, ok := device.(*v1beta1.Device)
+			if !ok {
+				klog.Warning("Failed to assert to CacheDevice type")
+				continue
+			}
+			deviceStates := &DeviceStatus{Status: cacheDevice.Status}
+			deviceStates.Status.State = msgState.Device.State
+			deviceStates.Status.LastOnline = msgState.Device.LastOnline
+
+			cacheDevice.Status.State = msgState.Device.State
+			cacheDevice.Status.LastOnline = msgState.Device.LastOnline
```

Comment:
I have some confusion here. The assignments of `deviceStates` and `cacheDevice` seem to be the same. Is it possible to use only one variable to describe them? Or to put it another way, can we directly use `cacheDevice.Status=deviceStates.Status` to replace it?